File: hw1/src/main/java/Parser.java
import java.util.InvalidPropertiesFormatException;

public class Parser {
    static String parseOperatorOrError(String operator) throws InvalidPropertiesFormatException {
        if ("+".equals(operator) ||
                "-".equals(operator) ||
                "*".equals(operator) ||
                "/".equals(operator)
        ) {

            return operator;
        }
        throw new InvalidPropertiesFormatException("Operator is not supported");
    }

    static int parseIntOrError(String val) {
        try {
            return Integer.parseInt(val);

        } catch (NumberFormatException e) {
            throw new NumberFormatException("Val is not int");
        }
    }

    static void countOfArgumentsChecker(String[] args) throws InvalidPropertiesFormatException {
        if (args.length != 3) {
            throw new InvalidPropertiesFormatException("Args format is: val1 +-/* val2");
        }
    }
}

File: hw2/src/main/kotlin/Parser.kt
import java.util.*

class Parser {
    fun parseIntOrError(num: String): Int {
        try {
            return Integer.parseInt(num)
        } catch (e: NumberFormatException) {
            throw NumberFormatException("Val is not int")
        }
    }

    fun parseOperatorOrError(operator: String): String {
        return when (operator) {
            "+" -> operator
            "-" -> operator
            "*" -> operator
            "/" -> operator
            else -> throw InvalidPropertiesFormatException("Operator is not supported")
        }

    }

    fun countOfArgumentsChecker(args: Array<String>) {
        if (args.size != 3) {
            throw InvalidPropertiesFormatException("Args format is: val1 +-/* val2")
        }
    }
}